### ISSUE TITLE
Storing varying datatypes as a time-series value

### ISSUE THREAD

**Issue by tjrocha** [usbr/Pisces#96]
The RWIS project is looking into storing non-numeric and/or file-based time-series data into the database. Think GIS polygons, PDF documents, engineering drawings, images, etc. Initial research shows that it is doable by either:
1. [storing the data as a BLOB datatype](http://forums.mysql.com/read.php?20,17671,27914) or by
2. storing links in the value field that point to the file in the DB server or a completely separate file server.

So the intent is to still store them as a time-series but instead of a double datatype value, we will be storing either a BLOB or a link. Most things I've read seem to suggest that option-2 above is better performance wise but option-1 benefits from the things that are typically associated with enterprise DB systems such as routine backups and security. Thoughts? Is this something that could have potential use for your other Pisces applications? 

@ktarbet @blounsbury @robertherrera

**Comment by ktarbet** [comment]
Yes, there is potential but we need a clean design to consider putting this in Pisces.

I've used blobs in the Hydrography application to store flow measurement raw data, rating tables, Excel, and zip files.  We also use blobs to store some reports, but not yet in Pisces. 

I'm interested in this idea of a blob instead of double.  However, what would Pisces show for the display?

Can you please give more details on the use case?  Or describe a clear scenario for this?  Are the PDF's site visit reports? Are the drawings related to the time series data?  What would the time stamp be used for in these cases?

**Comment by tjrocha** [comment]
> However, what would Pisces show for the display?

For now, nothing. Eventually, we can probably reconsider the idea of having an HTML browser as another tab so we can easily process/display these other datatypes via javascript libraries. Datatypes such as PDFs, images, etc.

> Can you please give more details on the use case? Or describe a clear scenario for this? Are the PDF's site visit reports? Are the drawings related to the time series data? What would the time stamp be used for in these cases?

A good example of this would be site visit reports for record keeping and sharing with the public and within Reclamation. If we were to try to incorporate the environmental field surveys into the RWIS project, we would need a way to store and retrieve site visit reports, files, and images from the DB. These reports would still be a time-series with a time stamp of when the site visit occurred. Think AgriMet calibration reports, I recall building an Access DB there in PN to store and track site calibration report data. Imagine if we can store that information into Pisces.

**Comment by ktarbet** [comment]
This could be in a class called  FileBlobSeries -- The Pisces table view could show links to the documents, when clicked would open PDF, XLS etc.   The file type may need to be stored in the flag column?  FileType=spreadsheet.
